src/components/profile/newIVAsModal/newIVAModal.tsx

```diff
@@ -0,0 +1,193 @@
+        setClickedButton("");
+        props.setShow(false);
+      } catch {}
+    }
+    setDisabledButton(false);
+    return;
+  };
+
+  return (
+    <Modal
+      show={props.show}
+      onHide={() => {
+        props.setShow(false);
+        setClickedButton("");
+        setPromptText("");
+      }}
+      size="lg"
+    >
+      <Modal.Header closeButton>
+        <Modal.Title>New verification address</Modal.Title>
+      </Modal.Header>
+      <Modal.Body>
+        <div className="mb-4">
+          <p>Please select one of the following verification addresses:</p>
+          <Row>
+            <Col>
+              <Button
+                className={
+                  clickedButton === "Phone"
+                    ? "w-100 bg-quaternary"
+                    : "w-100 bg-quinary"
+                }
+                onClick={() => {
+                  setClickedButton("Phone");
+                  setDisabledButton(false);
+                  setPromptText("mobile phone number");
+                }}
+              >
+                SMS
+              </Button>
+            </Col>
+            <Col>
+              <Button
+                className={
+                  clickedButton === "Fax"
+                    ? "w-100 bg-quaternary"
+                    : "w-100 bg-quinary"
+                }
+                onClick={() => {
+                  setClickedButton("Fax");
+                  setDisabledButton(false);
+                  setPromptText("fax number");
+                }}
+              >
+                Fax
+              </Button>
+            </Col>
+            <Col>
+              <Button
+                className={
+                  clickedButton === "PostalAddress"
+                    ? "w-100 bg-quaternary"
+                    : "w-100 bg-quinary"
+                }
+                onClick={() => {
+                  setClickedButton("PostalAddress");
+                  setDisabledButton(false);
+                  setPromptText("postal address");
+                }}
+              >
+                Letter
+              </Button>
+            </Col>
+            <Col>
+              <Button
+                className={
+                  clickedButton === "InPerson"
+                    ? "w-100 bg-quaternary"
+                    : "w-100 bg-quinary"
+                }
+                onClick={() => {
+                  setClickedButton("InPerson");
+                  setDisabledButton(false);
+                  setPromptText("Where can we meet you?");
+                }}
+              >
+                In-Person
+              </Button>
+            </Col>
+          </Row>
+        </div>
+        {clickedButton !== "" ? (
+          <form
+            onSubmit={(e) => {
+              e.preventDefault();
+              const target = e.target as HTMLFormElement;
+              handleSubmit(target.iva.value);
+            }}
+          >
+            <p>
+              {clickedButton === "In-Person"
```

Comment:
This does not work, since clickedButton is set to "InPerson" instead of "In-Person". You can use a typed version of useState to avoid such problems, something like `[clickedButton, setClickedButton] = useState<IvaType | null>(null)`.

Reply:
Done 👍